sqlite3_vendor/CMakeLists.txt

```
@@ -15,8 +15,8 @@ endif()
 ament_vendor(sqlite3_vendor
   SATISFIED ${SQLite3_FOUND}
   VCS_TYPE zip
-  VCS_URL https://www.sqlite.org/2022/sqlite-amalgamation-3370200.zip
-  VCS_VERSION sqlite-amalgamation-3370200
+  VCS_URL https://www.sqlite.org/2024/sqlite-amalgamation-3450300.zip
+  VCS_VERSION sqlite-amalgamation-3450300
```

The version from Ubuntu Noble is 3.45.1, so I think this should be:

```suggestion
  VCS_URL https://www.sqlite.org/2024/sqlite-amalgamation-3450100.zip
  VCS_VERSION sqlite-amalgamation-3450100
```

The `sqlite3.h` header in the ZIP file indeed shows version `3.45.1`.

---

I've made these changes in #1737